perfkitbenchmarker/pkb.py

```diff
@@ -198,17 +174,28 @@ def ListUnknownBenchmarks():
                 valid_benchmark_sets)
 
 
-def RunBenchmark(benchmark, collector):
+def RunBenchmark(benchmark, collector, sequence_number, total_benchmarks):
   """Runs a single benchmark and adds the results to the collector.
 
   Args:
     benchmark: The benchmark module to be run.
     collector: The SampleCollector object to add samples to.
+    sequence_number: The sequence number of when the benchmark was started
+      relative to the other benchmarks in the suite.
+    total_benchmarks: The total number of benchmarks in the suite.
   """
   benchmark_info = benchmark.GetInfo()
   benchmark_specification = None
   if not ValidateBenchmarkInfo(benchmark_info):
     return
+  benchmark_name = benchmark_info['name']
+
+  # Modify the logger prompt for messages logged within this function.
+  label_extension = log_util.CreateBenchmarkLabelExtension(
+      benchmark_name=benchmark_name if FLAGS.log_benchmark_name else None,
+      sequence_number=sequence_number if FLAGS.log_benchmark_sequence else None,
+      total_benchmarks=total_benchmarks)
+  label_extender = log_util.LabelExtender(label_extension)  # NOQA
```

Comment:
In response to your other comment about contextmanager, this will probably be removed.